src/gwt/src/org/rstudio/studio/client/common/rstudioapi/RStudioAPI.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * RStudioAPI.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+package org.rstudio.studio.client.common.rstudioapi;
+
+import org.rstudio.core.client.widget.HyperlinkLabel;
+import org.rstudio.core.client.widget.MessageDialog;
+import org.rstudio.core.client.widget.Operation;
+import org.rstudio.studio.client.RStudioGinjector;
+import org.rstudio.studio.client.application.events.EventBus;
+import org.rstudio.studio.client.common.GlobalDisplay;
+import org.rstudio.studio.client.common.rstudioapi.events.RStudioAPIShowDialogEvent;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.gwt.user.client.ui.HTML;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+@Singleton
+public class RStudioAPI implements RStudioAPIShowDialogEvent.Handler
+{
+   public RStudioAPI()
```

Comment:
Ah, I think I understand now. Since the RStudioAPI class will only ever be created by Guice, it'll take care of calling the `initialize()` method when the singleton RStudioAPI object is constructed, which will initialize its members. Does that sound correct?

For other objects that we create 'by hand' (e.g. explicitly with `new`), we typically need to manually call `injectMembers()` to ensure that `@Inject`-annotated methods are called as needed.